### PR TITLE
Add `recipe deselect` command

### LINKED ISSUE #262: [PE-D] No way to deselect a recipe
Before any recipes are added, other small inputs can be conducted when there are no recipes selected for editing, where the CLI will show `Decodex []`. However, once any recipes are added, the program will automatically have a recipe selected for editing. Hence `list recipes` can be called while editing a recipe which could mislead the user to think that recipes can be nested.'

Perhaps there could be a way to deselect recipes and return to the original state without having to invoke the `delete` command, as it seems to be the only way to deselect from all recipes now.

Just a small issue, but could potentially confuse users.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: alwinangys/ped#3

### PR DESCRIPTION
**Pull request by arraysius**
Fixes #262

### CODE REVIEW

File: src/main/java/decodex/recipes/RecipeManager.java [hunk 1]
```diff
@@ -124,6 +123,14 @@ public void selectRecipeForEditing(String name) throws RecipeManagerException {
         logger.fine(String.format("[RecipeManager] Selected recipe %s", name));
     }
 
+    /**
+     * Deselects the current editing recipe.
+     */
```
Comment (rizemon) [inline src/main/java/decodex/recipes/RecipeManager.java:128]: Perhaps change this to `Deselects the recipe currently being edited`?

**Review by rizemon — CHANGES_REQUESTED** [review]
Small changes, logic otherwise LGTM

**Review by rizemon — APPROVED** [review]
LGTM

File: src/main/java/decodex/commands/recipe/RecipeDeselectCommand.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public RecipeDeselectCommand() {
+        super();
+    }
+
+    @Override
+    public void run(DataManager dataManager, ModuleManager moduleManager, Ui ui, RecipeManager recipeManager,
+            Storage storage) throws CommandException, ModuleManagerException, DataManagerException, ModuleException,
+            RecipeException, RecipeManagerException, IOException {
+        String recipeName = recipeManager.getEditingRecipe().getName();
```
Comment (Kair0s3) [inline src/main/java/decodex/commands/recipe/RecipeDeselectCommand.java:32]: Looks good overall, just a minor suggestion. Perhaps, this line can be broken down?

**Review by Kair0s3 — CHANGES_REQUESTED** [review]

**Review by Kair0s3 — APPROVED** [review]
Looks
Good
To
Me

**Review by SeenFang — APPROVED** [review]
LGTM

**Pull request merged**